ssl/t1_lib.c

```diff
@@ -769,7 +774,7 @@ int tls12_check_peer_sigalg(const EVP_MD **pmd, SSL *s,
         /* Check compression and curve matches extensions */
         if (!tls1_set_ec_id(curve_id, &comp_id, EVP_PKEY_get0_EC_KEY(pkey)))
             return 0;
-        if (!s->server && !tls1_check_ec_key(s, curve_id, &comp_id)) {
+        if (!s->server && !tls1_check_ec_key(s, curve_id, &comp_id, 1)) {
```

Comment:
Would be nice to have a more detailed comment explaining the context. Are we choosing a signature algorithm to use for signing, or validating an algorithm used in a peer signature?  Or ...

Reply:
This is about checking a signature algorithm associated with a signature that has been sent to us by the peer is acceptable to us. I've expanded out the description of the tls12_check_peer_sigalg function to try and make this a bit clearer